test/realm_test.dart

```diff
@@ -1551,6 +1550,210 @@ Future<void> main([List<String>? args]) async {
+          originalRealm.close();
+
+          expect(File(pathCopy).existsSync(), isTrue);
+          final copiedRealm = getRealm(configCopy);
+          expect(copiedRealm.all<Car>().length, itemsCount);
+          expect(copiedRealm.isClosed, isFalse);
+          copiedRealm.close();
+        });
+      }
+    }
+  }
+
+  // writeCopy Sync to Sync realm
+  for (List<int>? sourceEncryptedKey in [null, generateEncryptionKey()]) {
+    for (List<int>? destinationEncryptedKey in [sourceEncryptedKey, generateEncryptionKey()]) {
+      final sourceEncryptedState = '${sourceEncryptedKey != null ? "encrypted " : ""}Sync';
+      final destinationEncryptedState =
+          'to ${destinationEncryptedKey != null ? "encrypted with ${sourceEncryptedKey != null && sourceEncryptedKey == destinationEncryptedKey ? "the same" : "different"} key " : ""}Sync';
+      final testDescription = '$sourceEncryptedState $destinationEncryptedState';
+      baasTest('Realm writeCopy Sync->Sync - $testDescription can be opened and synced', (appConfiguration) async {
+        final app = App(appConfiguration);
+        var user1 = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+        final originalConfig = Configuration.flexibleSync(user1, [Product.schema], encryptionKey: sourceEncryptedKey);
+        final originalRealm = getRealm(originalConfig);
+        var itemsCount = 100;
+        final productNamePrefix = generateRandomString(10);
+        await _addDataToAtlas(originalRealm, productNamePrefix, itemsCount: itemsCount);
+
+        var user2 = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+        final configCopy = Configuration.flexibleSync(user2, [Product.schema], encryptionKey: destinationEncryptedKey);
+        originalRealm.writeCopy(configCopy);
+        originalRealm.close();
+
+        expect(File(configCopy.path).existsSync(), isTrue);
+        // Check data in copied realm before synchronization
+        final disconnectedConfig = Configuration.disconnectedSync([Product.schema], path: configCopy.path, encryptionKey: destinationEncryptedKey);
+        final disconnectedCopiedRealm = getRealm(disconnectedConfig);
+        expect(disconnectedCopiedRealm.all<Product>().length, itemsCount);
+        disconnectedCopiedRealm.close();
+
+        // Sync copied realm to the server
+        final copiedRealm = getRealm(configCopy);
+        await _addSubscriptions(copiedRealm, productNamePrefix);
+        await copiedRealm.syncSession.waitForUpload();
+        await copiedRealm.syncSession.waitForDownload();
+        expect(copiedRealm.all<Product>().length, itemsCount);
+        expect(copiedRealm.isClosed, isFalse);
+        copiedRealm.close();
+
+        // Create another user's realm and download the data
+        var anotherUser = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+        final anotherUserRealm = getRealm(Configuration.flexibleSync(anotherUser, [Product.schema]));
```

Comment:
are we sure we are not opening the same realm here. isn't path required for what we are trying to do?

Reply:
When using `reuseCredentials: false` a new user is created with a new id. The path to the opened flexible synced realm with this user will contain the id of the new user.

Reply:
ok got it